test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -190,4 +221,241 @@ static public void testuninitializedVTClassHasNullDefaultValue() {
 		}
 		assertNull(myUnsafe.uninitializedDefaultValue(NeverInitialized.class));
 	}
+
+	@Test
+	static public void testNullObjGetValue() throws Throwable {
+		assertThrows(NullPointerException.class, () -> {
+			myUnsafe.getValue(null, 0, ValueTypeInt.class);
+		});
+	}
+
+	@Test
+	static public void testNullClzGetValue() throws Throwable {
+		assertThrows(NullPointerException.class, () -> {
+			myUnsafe.getValue(vtPoint, vtPointOffsetX, null);
+		});
+	}
+
+	@Test
+	static public void testNonVTClzGetValue() throws Throwable {
+		assertNull(myUnsafe.getValue(vtPoint, vtPointOffsetX, IntWrapper.class));
+	}
+
+	@Test
+	static public void testGetValuesOfArray() throws Throwable {
+		if (isFlatteningEnabled) {
+			ValueTypePoint2D p = myUnsafe.getValue(vtPointAry, vtPointAryOffset0, ValueTypePoint2D.class);
+			assertEquals(p.x.i, vtPointAry[0].x.i);
+			assertEquals(p.y.i, vtPointAry[0].y.i);
+			p = myUnsafe.getValue(vtPointAry, vtPointAryOffset1, ValueTypePoint2D.class);
+			assertEquals(p.x.i, vtPointAry[1].x.i);
+			assertEquals(p.y.i, vtPointAry[1].y.i);
+		}
```

Comment:
Do we need test cases if `isFlatteningEnabled` is false ?

Reply:
Yes, this should be possible if I use a ValueTypeInt array instead of a ValueTypePoint2D array (the lack of flattening in the ValueTypePoint2D objects make them very hard to use here, but ValueTypeInts should be fine).
There are also other tests (such as testGetValuesOfObject) where adding a case for when `isFlatteningEnabled` is false would also make sense. Should I do that for these other functions too?

Reply:
> Should I do that for these other functions too?

Yes, please.

Reply:
addressed in c83a184